tests/infer/test_valid_models.py

```diff
@@ -197,6 +197,47 @@ def guide():
     assert_error(model, guide, Elbo(), match='Multiple sample sites named')
 
 
+@pytest.mark.parametrize("has_rsample", [False, True])
+@pytest.mark.parametrize("Elbo", [Trace_ELBO, TraceGraph_ELBO, TraceEnum_ELBO, TraceTMC_ELBO])
+def test_set_has_rsample_ok(has_rsample, Elbo):
```

Comment:
can you please comment on the main goal of this and the following test?

Reply:
Done. Note this test isn't likely to fail now, but Hongseok and students may implement stricter checks for #2277, and those checks should take care to validate this model.